File: solutions/csharp/hamming/1/Hamming.cs
using System;using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException();
        }

        return firstStrand.Zip(secondStrand).Aggregate(0, (i, tuple) =>
        {
            if (tuple.First != tuple.Second)
            {
                return i + 1;
            }

            return i;
        });
    }
}

File: solutions/fsharp/hamming/1/Hamming.fs
module Hamming

let distance (strand1: string) (strand2: string): int option = 
    if (strand1.Length <> strand2.Length) 
    then None 
    else 
        List.zip (strand1 |> List.ofSeq) (strand2 |> List.ofSeq)
        |> List.sumBy (fun (c1, c2) -> if c1 = c2 then 0 else 1) 
        |> Some